src/HogWildTrainer.scala

```diff
@@ -1,22 +0,0 @@
-
-import cc.factorie.optimize.{ GradientOptimizer, Trainer, Example }
-import cc.factorie.la.SmartGradientAccumulator
-import cc.factorie.util.{ LocalDoubleAccumulator, Threading }
-import cc.factorie.model.WeightsSet
-
-class HogWildTrainer(val weightsSet: WeightsSet, val optimizer: GradientOptimizer, val nThreads: Int = Runtime.getRuntime.availableProcessors(), val maxIterations: Int = 3)
-  extends Trainer {
-
-  var iteration = 0
-  def processExample(e: Example): Unit = {
```

Comment:
Each `Example` knows how to feed forward itself and apply its own gradients, in the way that is unique to its model implementation